.github/workflows/ci.yml

```diff
@@ -68,3 +71,12 @@ jobs:
         with:
           components: clippy
       - run: cargo clippy -- -D warnings
+
+  detect-unused-dependencies:
+    name: Cargo Machete
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Machete
+        uses: bnjbvr/cargo-machete@main
```

Comment:
we could try using `udeps` which uses a more in-depth check. This is what upstream reth uses: https://github.com/paradigmxyz/reth/blob/main/.github/workflows/lint.yml#L159-L171

This should have fewer false positives.